### PR TITLE
Blog about Azure Provider 2.0

### LINKED ISSUE #2628: Blog about Azure Provider 2.0

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Fix #2628

TODOs:
- [ ] Meta image
- [ ] Finish updating examples to 2.0

### CODE REVIEW

File: content/blog/pulumi-azure-2-0/_index.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        update: "60m",
+    },
+});
+```
+
+You shouldn't need this ability often, but it can be a lifesaver when you do.
+
+## Getting Started with Migration
+
+To get started with Azure provider 2.0, update the versions in your package manager:
```
Comment (leezen) [inline content/blog/pulumi-azure-2-0/_index.md]: Would it make sense to use the language chooser here?

File: content/blog/pulumi-azure-2-0/_index.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+// Python: requirements.txt
+pulumi-azure>=2.0.0
+
+// Go: Gopkg.toml
+[constraint]]
+  version = "v2.0.0"
+  name = "github.com/pulumi/pulumi-azure"
+```
+
+There are quite a few breaking changes, so you may need to adjust your code before you would be able to run the program successfully again.
```
Comment (leezen) [inline content/blog/pulumi-azure-2-0/_index.md]: Anything notable we ought to call out? Or maybe link again to the TF upgrade guide?

File: content/blog/pulumi-azure-2-0/_index.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    adminPassword: password,
+    disablePasswordAuthentication: false,
+);
+```
+
+## Removing Deprecated Resources, Invokes, and Fields
+
+Azure Active Directory has its [own Pulumi provider](https://github.com/pulumi/pulumi-azuread/), so all the AD resources were now removed from the `AD` namespace of the Azure provider 2.0.
+
+A bunch of other resources, invokes, and fields were removed too, following the changes in the upstream provider. You can see the full list in [this upgrade guide](https://www.terraform.io/docs/providers/azurerm/guides/2.0-upgrade-guide.html).
```
Comment (spara) [inline content/blog/pulumi-azure-2-0/_index.md]:
```suggestion
A number of other resources, invokes, and fields were removed, following the changes in the upstream provider. You can see the full list in [this upgrade guide](https://www.terraform.io/docs/providers/azurerm/guides/2.0-upgrade-guide.html).
```

File: content/blog/pulumi-azure-2-0/_index.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+[Serverless as Callbacks](https://www.pulumi.com/blog/serverless-as-simple-callbacks-with-pulumi-and-azure-functions/) running on Azure Functions have bumped the default version of Azure Functions runtime to `~3` and `Node.js` to `~12`. As always, you can override the default to set the versions that you require.
+
+Callback Functions also moved from `ZipBlob` to `Blob` deploy as described above, so expect another replacement operation while upgrading.
+
+## Custom Timeouts
+
+Some types of resources take longer to create than others. For instance, provisioning a new Cosmos DB account may take from 10 minutes to over an hour depending on the number of geo locations and other factors.
+
+If you find that creation of a target resource times out, you can override the default timeout using resource options:
```
Comment (spara) [inline content/blog/pulumi-azure-2-0/_index.md]:
```suggestion
If the creation of a target resource times out, you can override the default timeout using resource options:
```

File: content/blog/pulumi-azure-2-0/_index.md (hunk 2 above)
Comment (spara) [inline content/blog/pulumi-azure-2-0/_index.md]:
```suggestion
There are quite a few breaking changes, so you may need to adjust your code before you can run the program successfully again.
```

File: content/blog/pulumi-azure-2-0/_index.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+[constraint]]
+  version = "v2.0.0"
+  name = "github.com/pulumi/pulumi-azure"
+```
+
+There are quite a few breaking changes, so you may need to adjust your code before you would be able to run the program successfully again.
+
+To help with this process, we upgraded all our Azure [templates](https://github.com/pulumi/templates/) and [examples](https://github.com/pulumi/examples/) to 2.0.
+
+If you want to stick with 1.x version for a while, please pin your version to `^1.0.0` or `1.14.0` in the package manager.
```
Comment (spara) [inline content/blog/pulumi-azure-2-0/_index.md]:
```suggestion
If you want to stay with the 1.x version, please pin your version to `^1.0.0` or `1.14.0` in the package manager.
```

File: content/blog/pulumi-azure-2-0/_index.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+If you still have an issue or a question, reach out to us on [Pulumi Community Slack](https://slack.pulumi.com/), and we'll help you through migration.
```
Comment (spara) [inline content/blog/pulumi-azure-2-0/_index.md]:
```suggestion
If you have an issues or questions, reach out to us on [Pulumi Community Slack](https://slack.pulumi.com/), and we'll help you through migration.
```

File: content/blog/pulumi-azure-2-0/_index.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
+---
+title: "Announcing Pulumi Azure Provider 2.0"
+authors: ["mikhail-shilkov"]
+tags: ["Azure"]
+date: "2020-03-11"
```
Comment (spara) [inline content/blog/pulumi-azure-2-0/_index.md]:
```suggestion
date: "2020-03-12"
```

File: content/blog/pulumi-azure-2-0/_index.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+meta_desc: "Announcing the 2.0 release of Pulumi Azure provider and what's new in it."
+---
+
+We are happy to announce the release of a new major version of Pulumi Azure provider. Pulumi Azure 2.0 is based on the [2.0 release](https://www.terraform.io/docs/providers/azurerm/guides/2.0-upgrade-guide.html) of the upstream provider and brings several improvements and breaking changes.
```
Comment (isaac-pulumi) [inline content/blog/pulumi-azure-2-0/_index.md]: "a new major version of **the** Pulumi Azure provider."

Comment (isaac-pulumi) [inline content/blog/pulumi-azure-2-0/_index.md]: Missing the article 'the':
"a new major version of **the** Pulumi Azure provider."

File: content/blog/pulumi-azure-2-0/_index.md [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+## Removing Deprecated Resources, Invokes, and Fields
+
+Azure Active Directory has its [own Pulumi provider](https://github.com/pulumi/pulumi-azuread/), so all the AD resources were now removed from the `AD` namespace of the Azure provider 2.0.
+
+A bunch of other resources, invokes, and fields were removed too, following the changes in the upstream provider. You can see the full list in [this upgrade guide](https://www.terraform.io/docs/providers/azurerm/guides/2.0-upgrade-guide.html).
+
+## Default Versions in Callback Functions
+
+[Serverless as Callbacks](https://www.pulumi.com/blog/serverless-as-simple-callbacks-with-pulumi-and-azure-functions/) running on Azure Functions have bumped the default version of Azure Functions runtime to `~3` and `Node.js` to `~12`. As always, you can override the default to set the versions that you require.
```
Comment (isaac-pulumi) [inline content/blog/pulumi-azure-2-0/_index.md]: Missing Article **the** below:
```suggestion
[Serverless as Callbacks](https://www.pulumi.com/blog/serverless-as-simple-callbacks-with-pulumi-and-azure-functions/) running on Azure Functions have bumped the default version of the Azure Functions runtime to `~3` and `Node.js` to `~12`. As always, you can override the default to set the versions that you require.
```

File: content/blog/pulumi-azure-2-0/_index.md (hunk 2 above)
Comment (mikhailshilkov) [reply content/blog/pulumi-azure-2-0/_index.md]: Added another link to the upgrade guide. Does it look better now?

File: content/blog/pulumi-azure-2-0/_index.md (hunk 1 above)
Comment (mikhailshilkov) [reply content/blog/pulumi-azure-2-0/_index.md]: Great idea!

File: content/blog/pulumi-azure-2-0/_index.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+    computerName: "hostname",
+    adminUsername: username,
+    adminPassword: password,
+    disablePasswordAuthentication: false,
+);
+```
+
+## Removing Deprecated Resources, Invokes, and Fields
+
+Azure Active Directory has its [own Pulumi provider](https://github.com/pulumi/pulumi-azuread/), so all the AD resources were now removed from the `AD` namespace of the Azure provider 2.0.
```
Comment (isaac-pulumi) [inline content/blog/pulumi-azure-2-0/_index.md]:
```suggestion
Azure Active Directory now has its [own Pulumi provider](https://github.com/pulumi/pulumi-azuread/), so all the AD resources were removed from the `AD` namespace of the Azure Provider 2.0.
```

**Pull request merged**